### PR TITLE
[W5.11][W13-1]Teh Nian Fei

### PR DESCRIPTION
**Pull request by nianfei97**
Ready for review

### CODE REVIEW

**Comment by cjinting-nus** [comment]
The function has been written in the OOP-standard, with appropriate comments added to the code and the userguide. The tests also have been appropriately updated to reflect the expected output. Nice one! One question though, is there a reason as to why the ArrayList instead of a Set?

File: test/java/seedu/addressbook/commands/FindCommandTest.java [hunk 1]
```diff
@@ -25,8 +25,8 @@ public void execute() throws IllegalValueException {
         //same word, same case: matched
         assertFindCommandBehavior(new String[]{"Amy"}, Arrays.asList(td.amy));
 
-        //same word, different case: not matched
-        assertFindCommandBehavior(new String[]{"aMy"}, Collections.emptyList());
+        //same word, different case: matched
+        assertFindCommandBehavior(new String[]{"aMy"}, Arrays.asList(td.amy));
```
Comment (okkhoy) [inline test/java/seedu/addressbook/commands/FindCommandTest.java:29]: 👍 for updating the Junit tests!

**Review by okkhoy — APPROVED** [review]
This enhancement surely gives you a tick mark on the graded LO.
However, not sure if the intended objective of being able to work with an intermediate size codebase was achieved.

**Comment by okkhoy** [comment]
@nianfei97 
Some comments added. Please take note and close the pull request.

**Pull request closed without merging**